src/components/auth/Login.jsx

```diff
@@ -16,33 +43,64 @@ export function Login() {
             <input
               type="text"
               className={`full-width input-line ${auth["input-line"]} `}
+              onChange={(e) => {
+                loginInfoDispatch({ type: "EMAIL", payload: e.target.value });
+              }}
             />
           </label>
 
           <label className={`flex-col gap-sm full-width`}>
             Enter password
             <div className="flex-row">
               <input
-                type="password"
+                type={isPwdVisible ? "text" : "password"}
                 className={`full-width input-line ${auth["input-line"]}  `}
+                onChange={(e) => {
+                  loginInfoDispatch({ type: "PWD", payload: e.target.value });
+                }}
               />
-              <span className="pwd-eye">
+              <span
+                className="pwd-eye"
+                onClick={() => {
+                  setIsPwdVisible((p) => !p);
+                }}
+              >
                 <i className="fas fa-eye-slash"></i>
               </span>
             </div>
           </label>
 
           <div className={`flex-row spc-btwn full-width`}>
             <label>
-              <input type="checkbox" /> Remember me
+              <input
+                type="checkbox"
+                onChange={() => {
+                  loginInfoDispatch({ type: "REMEMBER" });
+                }}
+              />{" "}
+              Remember me
             </label>
 
             <a href="#" className="footer-link">
               Forgot your password?
             </a>
           </div>
 
-          <button className="cart-btn full-width">Login</button>
+          <button
+            className="cart-btn full-width"
+            onClick={async () => {
+              try {
+                let response = await postUserLogin(loginInfo);
+                setError("");
+                userDataDispatch({ type: "LOGIN", payload: response.data });
+                navigate("/products");
+              } catch (err) {
+                setError(err.response.data.message);
+              }
```

Comment:
Any specific reason for inline functions ?. It would be better if function is written separately outside of return and it will increase the code readability.

Reply:
i just find them easier to write, will change them .